packages/utils/src/stableMath.ts

```diff
@@ -0,0 +1,188 @@
+      continue;
+    }
+
+    _x = balances[i];
+
+    S_ = S_.plus(_x);
+    c = c.multipliedBy(invariant).dividedBy(_x.multipliedBy(numTokens));
+  }
+
+  c = c.multipliedBy(invariant).dividedBy(Ann.multipliedBy(numTokens));
+  const b = S_.plus(invariant.dividedBy(Ann));
+
+  let y_prev = BIG_ZERO;
+  let y = invariant;
+
+  for (let i = 0; i < 255; i++) {
+    y_prev = y;
+    y = y.pow(2).plus(c).div(y.multipliedBy(2).plus(b).minus(invariant));
+
+    if (y.minus(y_prev).abs().lte(BIG_ONE)) {
+      break;
+    }
+  }
+
+  return y;
+};
+
+/**
+ * Computes how many tokens can be taken out of a pool if `tokenAmountIn` are sent, given the current balances.
+ *
+ * @param amp - The amplification parameter
+ * @param balances - Array of pool balances
+ * @param tokenIndexIn - Input token Index of the pool array
+ * @param tokenIndexOut - Output token Index of the pool array
+ * @param tokenAmountIn - Input Amount of token
+ * @returns BigNumberish string
+ */
+export const getAmountsOut = (
+  amp: string,
+  balances: string[],
+  tokenIndexIn: number,
+  tokenIndexOut: number,
+  tokenAmountIn: string,
+): string => {
+  /**************************************************************************************************************
+  // getAmountsOut token x for y - polynomial equation to solve                                                //
+  // ay = amount out to calculate                                                                              //
+  // by = balance token out                                                                                    //
+  // y = by - ay (finalBalanceOut)                                                                             //
+  // D = invariant                                               D                     D^(n+1)                 //
+  // A = amplification coefficient               y^2 + ( S - ----------  - D) * y -  ------------- = 0         //
+  // n = number of tokens                                    (A * n^n)               A * n^2n * P              //
+  // S = sum of final balances but y                                                                           //
+  // P = product of final balances but y                                                                       //
+  **************************************************************************************************************/
+
+  const convertedBalances = balances.map((balance) => new BigNumber(balance));
+  const convertedAmp = new BigNumber(amp);
+
+  const lastInvariant = _calculateInvariant(convertedAmp, convertedBalances);
+
+  convertedBalances[tokenIndexIn] = convertedBalances[tokenIndexIn].plus(tokenAmountIn);
+
+  const finalBalanceOut: BigNumber = _getTokenBalanceGivenInvariantAndAllOtherBalances(
+    convertedAmp,
+    convertedBalances,
+    lastInvariant,
+    tokenIndexOut,
+  );
+
+  return convertedBalances[tokenIndexOut].minus(finalBalanceOut).minus(BIG_ONE).toFixed(0).toString();
```

Comment:
ahh nice, disregard above comment since this should convert to an integer string!

Reply:
:)